src/io/platform/f4xx/PWMf4xx.cpp

```diff
@@ -0,0 +1,213 @@
+    case Pin::PB_6:
+        *instance = TIM4;
+        *channel = TIM_CHANNEL_1;
+        *alternateFunction = GPIO_AF2_TIM4;
+        break;
+    case Pin::PB_7:
+        *instance = TIM4;
+        *channel = TIM_CHANNEL_2;
+        *alternateFunction = GPIO_AF2_TIM4;
+        break;
+    case Pin::PB_8:
+        *instance = TIM4;
+        *channel = TIM_CHANNEL_3;
+        *alternateFunction = GPIO_AF2_TIM4;
+        break;
+    case Pin::PB_9:
+        *instance = TIM4;
+        *channel = TIM_CHANNEL_4;
+        *alternateFunction = GPIO_AF2_TIM4;
+        break;
+    default:
+        *instance = NULL;
+        *channel = -1;
+        *alternateFunction = -1;
+    }
+}
+
+PWMf4xx::PWMf4xx(Pin pin) : PWM(pin) {
+    TIM_TypeDef* instance;
+    uint32_t alternateFunction;
+    getInstance(pin, &instance, &halTIMChannelID, &alternateFunction);
+
+    if (instance == TIM1) {
+        __HAL_RCC_TIM1_CLK_ENABLE();
+    } else if (instance == TIM2) {
+        __HAL_RCC_TIM2_CLK_ENABLE();
+    } else if (instance == TIM3) {
+        __HAL_RCC_TIM3_CLK_ENABLE();
+    } else if (instance == TIM4) {
+        __HAL_RCC_TIM4_CLK_ENABLE();
+    }
+
+    TIM_BreakDeadTimeConfigTypeDef sBreakDeadTimeConfig = {0};
+
+    halTIM.Instance = instance;
+    halTIM.Init.Prescaler = 0;
+    halTIM.Init.CounterMode = TIM_COUNTERMODE_UP;
+    halTIM.Init.Period = 0;
+    halTIM.Init.ClockDivision = TIM_CLOCKDIVISION_DIV1;
+    halTIM.Init.RepetitionCounter = 0;
+    halTIM.Init.AutoReloadPreload = TIM_AUTORELOAD_PRELOAD_ENABLE;
+    HAL_TIM_Base_Init(&halTIM);
+
+    HAL_TIM_PWM_Init(&halTIM);
+
+    sBreakDeadTimeConfig.OffStateRunMode = TIM_OSSR_DISABLE;
+    sBreakDeadTimeConfig.OffStateIDLEMode = TIM_OSSI_DISABLE;
+    sBreakDeadTimeConfig.LockLevel = TIM_LOCKLEVEL_OFF;
+    sBreakDeadTimeConfig.DeadTime = 0;
+    sBreakDeadTimeConfig.BreakState = TIM_BREAK_DISABLE;
+    sBreakDeadTimeConfig.BreakPolarity = TIM_BREAKPOLARITY_HIGH;
+    sBreakDeadTimeConfig.BreakFilter = 0;
+    sBreakDeadTimeConfig.AutomaticOutput = TIM_AUTOMATICOUTPUT_DISABLE;
+    HAL_TIMEx_ConfigBreakDeadTime(&halTIM, &sBreakDeadTimeConfig);
+
+    // Setup GPIO pin for PMW
+    GPIO_InitTypeDef gpioInit = {0};
+    Pin myPins[] = {pin};
+    uint8_t numOfPins = 1;
+
+    GPIOf4xx::gpioStateInit(&gpioInit, myPins, numOfPins, GPIO_MODE_AF_PP,
+                            GPIO_NOPULL, GPIO_SPEED_FREQ_LOW,
+                            alternateFunction);
+}
+
+void PWMf4xx::setDutyCycle(uint32_t dutyCycle) {
+    this->dutyCycle = dutyCycle;
+
+    TIM_OC_InitTypeDef sConfigOC = {0};
+    sConfigOC.OCMode = TIM_OCMODE_PWM1;
+    sConfigOC.Pulse = dutyCycle * halTIM.Init.Period / 100;
+    sConfigOC.OCPolarity = TIM_OCPOLARITY_HIGH;
+    sConfigOC.OCNPolarity = TIM_OCNPOLARITY_HIGH;
+    sConfigOC.OCFastMode = TIM_OCFAST_DISABLE;
+    sConfigOC.OCIdleState = TIM_OCIDLESTATE_RESET;
+    sConfigOC.OCNIdleState = TIM_OCNIDLESTATE_RESET;
+
+    HAL_TIM_PWM_Stop(&halTIM, halTIMChannelID);
```

Comment:
Can you add this to PWMf3xx.cpp as well? Be sure to test that things still work after the change